test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -38,27 +40,26 @@
 
 (deftest test-error-during-init
   (testing
-      (str "If there as an exception while putting a JRubyPuppet instance in "
-           "the pool the application should shut down.")
+   (str "If there as an exception while putting a JRubyPuppet instance in "
+        "the pool the application should shut down.")
     (logging/with-test-logging
-      (with-redefs [jruby-internal/create-pool-instance!
-                    (fn [& _] (throw (Exception. "42")))]
-                   (let [got-expected-exception (atom false)]
-                     (try
-                       (bootstrap/with-app-with-config
-                         app
-                         default-services
-                         (jruby-service-test-config 1)
-                         (tk/run-app app))
-                       (catch Exception e
-                         (let [cause (stacktrace/root-cause e)]
-                           (is (= (.getMessage cause) "42"))
-                           (reset! got-expected-exception true))))
-                     (is (true? @got-expected-exception)
-                                "Did not get expected exception."))
-                   (is (logged?
-                         #"^shutdown-on-error triggered because of exception!"
-                                :error))))))
+     (with-redefs [jruby-internal/create-pool-instance!
+                   (fn [& _] (throw (Exception. "42")))]
+       (let [got-expected-exception (atom false)]
+         (try
+           ; We use this instead of with-app-with-config because with-app-with-config has an implicit stop
```

Comment:
Once the HUP changes were in we started seeing a double-stop, which broke the logging check. However, now that the partial-pool initialization throws an error instead of just logging a warning, we still have a double stop, so this could probably be reverted to the old test with the logged? check removed.